src/core/models/outputs/output_foundry.h

```diff
@@ -6,22 +6,104 @@
 
 #include <stdint.h>
 
+#include "core/models/outputs/feat_blocks.h"
 #include "core/models/outputs/native_tokens.h"
+#include "core/utils/uint256.h"
+
+// Token tag length in bytes
+#define TOKEN_TAG_BYTES_LEN 12
+
+/**
+ * @brief Token schemes
+ *
+ */
+typedef enum {
+  SIMPLE_TOKEN_SCHEME = 0  // For now, only token scheme 0 is supported.
+} token_scheme_e;
 
 /**
  * @brief An output type which controls the supply of user defined native tokens.
  *
  */
 typedef struct {
-  void* address;                  ///< The alias controlling this foundry
-  uint64_t amount;                ///< The amount of IOTA tokens held by this output
-  native_tokens_t native_tokens;  ///< The native tokens held by this output
-  uint32_t serial;                ///< The serial number of the foundry
-  void* token_tag;                ///< The tag which is always the last 12 bytes of the tokens generated by this foundry
-  void* circulating_supply;       ///< The circulating supply of tokens controlled by this foundry
-  void* max_supply;               ///< The maximum supply of tokens controlled by this foundry
-  void* token_scheme;             ///< The token scheme used by this foundry
-  void* feature_blocks;           ///< The feature flocks which modulate the constrants on this output
+  address_t* address;                     ///< The alias controlling this foundry
+  uint64_t amount;                        ///< The amount of IOTA tokens held by this output
+  native_tokens_t* native_tokens;         ///< The native tokens held by this output
+  uint32_t serial;                        ///< The serial number of the foundry
+  byte_t token_tag[TOKEN_TAG_BYTES_LEN];  ///< The tag which is always the last 12 bytes of the tokens generated by this
+                                          ///< foundry
+  uint256_t circ_supply;                  ///< The circulating supply of tokens controlled by this foundry
+  uint256_t max_supply;                   ///< The maximum supply of tokens controlled by this foundry
```

Comment:
Should those two elements be pointers? Each is 32 bytes long.

Reply:
As per discussion with Matjaz, eventhough uint256_t circ_supply and max_supply can store 32 bytes data, it is better to declare them as pointers considering the larger size of the variables which will lead to higher stack usage.